### PR TITLE
cleanup: Simplify deepcopy-gen code generation

### PR DESCRIPTION
**Pull request by jwcesign**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
Simplify deepcopy-gen code generation

**Which issue(s) this PR fixes**:
Part of https://github.com/karmada-io/karmada/issues/4230

**Special notes for your reviewer**:
I removed the `--output-package` parameter as it has no effect, the generation works fine.

The CI verification now checks `pkg` and `examples/customresourceinterpreter/apis`, but does not include `operation` parts. I will add them in another PR.

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

File: hack/update-codegen.sh [hunk 1]
```diff
@@ -34,62 +34,18 @@ echo "Generating with deepcopy-gen"
...
-  --output-file-base=zz_generated.deepcopy
-deepcopy-gen \
-  --go-header-file hack/boilerplate/boilerplate.go.txt \
   --input-dirs=github.com/karmada-io/karmada/pkg/apis/networking/v1alpha1 \
-  --output-package=github.com/karmada-io/karmada/pkg/apis/networking/v1alpha1 \
-  --output-file-base=zz_generated.deepcopy
-deepcopy-gen \
-  --go-header-file hack/boilerplate/boilerplate.go.txt \
   --input-dirs=github.com/karmada-io/karmada/examples/customresourceinterpreter/apis/workload/v1alpha1 \
-  --output-package=github.com/karmada-io/karmada/examples/customresourceinterpreter/apis/workload/v1alpha1 \
```
Comment (XiShanYongYe-Chang) [inline hack/update-codegen.sh:72]: This item has a different output path from the other items, and may require separate handling. The operator package is also in the same situation.

Comment (XiShanYongYe-Chang) [reply hack/update-codegen.sh:72]: The CI doesn't echo error, maybe we need to add the verify script for `examples` and `operator`.

Comment (jwcesign) [reply hack/update-codegen.sh:72]: I removed the `--output-package` parameter as it has no effect.
The CI verification now checks pkg and examples/customresourceinterpreter/apis, but does not include operation parts. I will add them in another PR.

**Review by XiShanYongYe-Chang — COMMENTED** [review]
LGTM

File: hack/update-codegen.sh (hunk 1 above)
Comment (RainbowMango) [reply hack/update-codegen.sh:72]: > I removed the --output-package parameter as it has no effect.

I'm surprised, why no effect?

> The CI verification now checks pkg and examples/customresourceinterpreter/apis, but does not include operation parts.

What do you mean?

Comment (jwcesign) [reply hack/update-codegen.sh:72]: > I'm surprised, why no effect?

I reviewed the code of deepcopy-gen and found that it only uses `--output-base` to specify the output path.
Passing parameters.
image

Write to file.
image


related code: https://github.com/kubernetes/code-generator

The testing result:

image

Comment (jwcesign) [reply hack/update-codegen.sh:72]: > What do you mean?

We only verify directory `/pkg` and `/examples/customresourceinterpreter/apis`
https://github.com/karmada-io/karmada/blob/bb1d2e8f6feaf8881931129ef4abf8e573a9ccd4/hack/verify-codegen.sh#L9


But there is https://github.com/karmada-io/karmada/blob/master/operator/pkg/apis/operator/v1alpha1/zz_generated.deepcopy.go

We forgot to verify this file.

Comment (RainbowMango) [reply hack/update-codegen.sh:72]: Can we first fix this issue? 

I need some time to investigate why `code-generater` does not support the `--output-base`, and if we have a chance to use it in the future.

Comment (jwcesign) [reply hack/update-codegen.sh:72]: Knative also does not utilize this parameter: https://github.com/knative/serving/blob/936ecbdb547d896d1ef260c17edb4131cfd1d2ae/hack/update-codegen.sh#L83

**Comment by jwcesign** [comment]
/cc @RainbowMango 
Any additional comments?

**Comment by RainbowMango** [comment]
Not yet. Will do.

**Comment by RainbowMango** [comment]
Starting from v1.30, the code generator removed the `--output-package` and required the `--output-file`.
So, we don't need this PR anymore. The adoption code is included in #5146. 
But still thanks for your effort. 
/close

**Pull request closed without merging**